inference-engine/thirdparty/clDNN/src/gpu/loop_gpu.cpp

```diff
@@ -208,9 +141,19 @@ struct loop_gpu : typed_primitive_impl<loop> {
             concat_output.restore_concatenated_mem();
         }
 
+        // update num_iterations (actual number of iterations)
+        int64_t actual_iterations = 0;
+        if (node.is_current_iteration_used()) {
+            const auto& backedge_mapping = instance.get_current_iteration_backedge_mapping();
+            auto current_iteration_mem = backedge_mapping.from_primitive->output_memory_ptr();
+            actual_iterations = loop_node::read_scalar_value(current_iteration_mem, stream);
+        } else {
+            actual_iterations = current_iteration_idx;
+        }
+
         const primitive_id& num_iteration_id = node.get_num_iteration_id();
         memory::ptr num_actual_iterations_mem = outer_network.get_primitive(num_iteration_id)->output_memory_ptr();
-        write_scalar_value(num_actual_iterations_mem, stream, current_iteration);
+        loop_node::write_scalar_value(num_actual_iterations_mem, stream, actual_iterations);
```

Comment:
I think this is not needed when the loop is fixed length with trip count.

Reply:
I fixed it. thanks.